src/process.py

```diff
@@ -0,0 +1,98 @@
+import time
+
+class Process:
+    def __init__(self, remainingIterations, size, sleepInterval):
+        self.remainingIterations = remainingIterations
+        self.sleepInterval = sleepInterval
+        self.size = size
+
+class CPU:
+    _instance = None
+
+    def __new__(cnt):
+        if not cnt._instance:
+            cnt._instance = super().__new__(cnt)
+            cnt._instance.os = OS()
+        return cnt._instance
+
+    def run(self):
+        while self.os.hasProcesses():
+            proc = self.os.getProcess()
+            if not proc:
+                time.sleep()
+            else:
+                while proc.remainingIterations > 0:
+                    
+                    for i in range(proc.size):
+                        continue
+                
+                    proc.remainingIterations -= 1
+                    if proc.sleepInterval > 0: #ako je sleepInterval is 0, no need to sleep
+                        self.os.sleep(proc.sleepTime, proc)
+                else:
+                    self.os.finishProcess(proc)
+
+class CPUScheduler:
+    def __init__(self):
+        self.processes = [] #za sad je tu al kad dodamo scheduler, ovde ce biti prazno
+        numberOfProcesses = 0 
+
+    def getProcess(self):
+        if self.processes:
+            return self.processes.pop(0)
+        return None
+
+    def putProcess(self, proc):
+        self.processes.append(proc)
+        self.numberOfProcesses += 1
+
+class OS:
+    _instance = None
+
+    def __new__(cnt):
+        if not cnt._instance:
+            cnt._instance = super().__new__(cnt)
+            cnt._instance.cpu = CPU()
+        return cnt._instance
+
+    def __init__(self):
+        self.blockedProcesses = []
+        self.numberOfProcesses = 0
+        self.cpuScheduler = CPUScheduler()
+
+    def createProcess(self, proc):
+        self.cpuScheduler.putProcess(proc)
+
+    def hasProcesses(self):
+        return self.numberOfProcesses > 0
+    
+    def getProcess(self):
+        for proceess in self.blockedProcesses:
+            if proceess.wakeTime <= time.time():
+                self.blockedProcesses.remove(proceess)
+                self.numberOfProcesses -= 1
```

Comment:
number of proceses nije broj blokiranih nego broj blokiranih i ready
inkrementira se on createProcess a dekrementira se u finishProcess